### PR TITLE
Upgrade to lyon 1.0

### PR DESCRIPTION
**Pull request by ogoffart**

### CODE REVIEW

File: internal/core/animations.rs [hunk 1]
```diff
@@ -295,8 +294,6 @@ pub fn easing_curve(curve: &EasingCurve, value: f32) -> f32 {
                 ctrl2: (*c, *d).into(),
                 to: (1., 1.).into(),
             };
-            #[cfg(feature = "std")]
-            let curve = curve.assume_monotonic();
```
Comment (tronical) [inline internal/core/animations.rs:299]: I hope this isn’t needed

File: internal/core/layout.rs [hunk 2]
```diff
@@ -625,27 +625,14 @@ pub fn solve_path_layout(
...
-            BezierSegment::Linear(line_segment) => line_segment.length(),
-            BezierSegment::Quadratic(quadratic_segment) => {
-                quadratic_segment.approximate_length(tolerance)
-            }
-            BezierSegment::Cubic(cubic_segment) => cubic_segment.approximate_length(tolerance),
-        })
-        .collect();
-
-    let path_length: f32 = segment_lengths.iter().sum();
+    let path_length = lyon_algorithms::length::approximate_length(path_iter.iter(), tolerance);
```
Comment (tronical) [inline internal/core/layout.rs:635]: Nice :)

**Review by tronical — APPROVED** [review]

File: internal/core/animations.rs (hunk 1 above)
Comment (ogoffart) [reply internal/core/animations.rs:299]: No, this was not needed for my implementation which I copied, iirc.

File: internal/core/layout.rs (hunk 2 above)
Comment (ogoffart) [reply internal/core/layout.rs:635]: Before, this was computing the length and adding the stuff to the vector. Now this is done in two steps

**Pull request merged**